qiskit_machine_learning/kernels/statevector_kernel.py

```diff
@@ -0,0 +1,191 @@
+                in the dataset, then the kernel will try to adjust the feature map to reflect the
+                number of features.
+            evaluate_duplicates: Defines a strategy how kernel matrix elements are evaluated if
+               duplicate samples are found. Possible values are:
+
+                    - ``all`` means that all kernel matrix elements are evaluated, even the diagonal
+                      ones when training. This may introduce additional noise in the matrix.
+                    - ``off_diagonal`` when training the matrix diagonal is set to `1`, the remaining
+                      elements are fully evaluated, e.g., for two identical samples in the
+                      dataset. When inferring, all elements are evaluated. This is the default
+                      value.
+                    - ``none`` when training the diagonal is set to `1` and if two identical samples
+                      are found in the dataset the corresponding matrix element is set to `1`.
+                      When inferring, matrix elements for identical samples are set to `1`.
+            statevector_type: The type of Statevector that will be instantiated using the
+                ``feature_map`` quantum circuit and used to compute the fidelity kernel. This type
+                should inherit from and defaults to :class:`~qiskit.quantum_info.Statevector`.
+
+        Raises:
+            ValueError: When unsupported value is passed to `evaluate_duplicates`.
+        """
+        super().__init__(feature_map=feature_map)
+
+        eval_duplicates = evaluate_duplicates.lower()
+        if eval_duplicates not in ("all", "off_diagonal", "none"):
+            raise ValueError(
+                f"Unsupported value passed as evaluate_duplicates: {evaluate_duplicates}"
+            )
+        self._evaluate_duplicates = eval_duplicates
+
+        self._statevector_type = statevector_type
+
+        self._statevector_cache: dict[tuple[float, ...], np.ndarray] = {}
+
+    def evaluate(self, x_vec: np.ndarray, y_vec: np.ndarray | None = None) -> np.ndarray:
+
+        x_vec, y_vec = self._validate_input(x_vec, y_vec)
+
+        # determine if calculating self inner product
+        is_symmetric = True
+        if y_vec is None:
+            y_vec = x_vec
+        elif not np.array_equal(x_vec, y_vec):
+            is_symmetric = False
+
+        kernel_shape = (x_vec.shape[0], y_vec.shape[0])
+
+        x_svs = [self._get_statevector(x) for x in x_vec]
+        y_svs = [self._get_statevector(y) for y in y_vec]
+
+        kernel_matrix = np.ones(kernel_shape)
+        for i, x in enumerate(x_svs):
+            for j, y in enumerate(y_svs):
+                if self._is_trivial(i, j, x, y, is_symmetric):
+                    continue
+                kernel_matrix[i, j] = self._compute_kernel_element(x, y)
+
+        # due to truncation and rounding errors we may get complex numbers
+        kernel_matrix = np.real(kernel_matrix)
```

Comment:
Is this still the case? Per typehint (and from what numpy returns) it seems that the kernel can only have floats as values 😄